src/backend/distributed/metadata/node_metadata.c

```diff
@@ -753,13 +754,662 @@ PgDistTableMetadataSyncCommandList(void)
+			scanEnded = true;
+			MemoryContextSwitchTo(metadataSyncContext);
+			continue;
+		}
+		MemoryContextSwitchTo(metadataSyncContext);
+
+		Form_pg_dist_object pg_dist_object = (Form_pg_dist_object) GETSTRUCT(
+			nextTuple);
+
+		objectAddress = palloc0(sizeof(ObjectAddress));
+		ObjectAddressSubSet(*objectAddress, pg_dist_object->classid,
+							pg_dist_object->objid,
+							pg_dist_object->objsubid);
+
+		bool distributionArgumentIndexIsNull = false;
+		Datum distributionArgumentIndexDatum =
+			heap_getattr(nextTuple,
+						 Anum_pg_dist_object_distribution_argument_index,
+						 tupleDesc,
+						 &distributionArgumentIndexIsNull);
+		int32 distributionArgumentIndex = DatumGetInt32(
+			distributionArgumentIndexDatum);
+
+		bool colocationIdIsNull = false;
+		Datum colocationIdDatum =
+			heap_getattr(nextTuple,
+						 Anum_pg_dist_object_colocationid,
+						 tupleDesc,
+						 &colocationIdIsNull);
+
+		bool forceDelegationIsNull = false;
+		Datum forceDelegationDatum =
+			heap_getattr(nextTuple,
+						 Anum_pg_dist_object_force_delegation,
+						 tupleDesc,
+						 &forceDelegationIsNull);
+
+		if (distributionArgumentIndexIsNull)
+		{
+			distArgumentIndex = INVALID_DISTRIBUTION_ARGUMENT_INDEX;
+		}
+		else
+		{
+			distArgumentIndex = distributionArgumentIndex;
+		}
+
+		if (colocationIdIsNull)
+		{
+			colocationId = INVALID_COLOCATION_ID;
+		}
+		else
+		{
+			colocationId = DatumGetInt32(colocationIdDatum);
+		}
+
+		if (forceDelegationIsNull)
+		{
+			forceDelegation = NO_FORCE_PUSHDOWN;
+		}
+		else
+		{
+			forceDelegation = DatumGetBool(forceDelegationDatum);
+		}
+
+		char *workerMetadataUpdateCommand =
+			MarkObjectsDistributedCreateCommand(list_make1(objectAddress),
+												list_make1_int(distArgumentIndex),
+												list_make1_int(colocationId),
+												list_make1_int(forceDelegation));
+		List *commandList = list_make1(workerMetadataUpdateCommand);
+		commandList = lcons(DISABLE_DDL_PROPAGATION, commandList);
+		commandList = lappend(commandList, ENABLE_DDL_PROPAGATION);
+		SendMetadataCommandListToWorkerListInCoordinatedTransaction(nodesWithMetadata,
+																	CurrentUserName(),
+																	commandList);
+	}
+
+	systable_endscan_ordered(scanDesc);
+	index_close(indexRelation, AccessShareLock);
+	table_close(relation, AccessShareLock);
+	list_free(nodesWithMetadata);
+
+	MemoryContextSwitchTo(oldContext);
+	MemoryContextDelete(metadataSyncContext);
+}
+
+
+/*
+ * SendColocationMetadataCommands sends colocation metadata within metadata
+ * connection.
+ */
+void
+SendColocationMetadataCommands(List *nodeList)
+{
+	List *nodesWithMetadata = NIL;
+	WorkerNode *workerNode = NULL;
+	foreach_ptr(workerNode, nodeList)
+	{
+		if (NodeIsPrimary(workerNode) && !NodeIsCoordinator(workerNode))
+		{
+			nodesWithMetadata = lappend(nodesWithMetadata, workerNode);
+		}
+	}
+
+	if (nodesWithMetadata == NIL)
+	{
+		return;
+	}
+
+	List *colocationGroupSyncCommandList = ColocationGroupCreateCommandList();
```

Comment:
In the future, we'll have much more colocation groups. Can't we make this similar to others?